contributor-guide/modules/ROOT/pages/testing/intro.adoc

```diff
@@ -1,29 +1,95 @@
 = Introduction to Testing
 :navtitle: Introduction
 
-The Boost libraries are intended to be both reliable and portable. Every experienced programmer knows that means each library must be tested against a suitable number of test cases, on a range of platforms and compilers, and then tested again (_regression tested_) every time a change is made and also before every public release.
+The Boost libraries are intended to be both reliable and portable. Every experienced programmer knows that means each library must be tested against a suitable number of test cases, on a range of platforms and compilers, and then tested again (_regression tested_) every time a change is made. The regression tests are run again before every public release.
 
 There are three broad dimensions to Boost library testing:
 
 . The actual test codes - unit tests, regression tests, new feature and bug tests. Refer to xref:testing/writing-tests.adoc[].
 . Tests that are integrated with the primary build systems, xref:build-systems/b2.adoc[] and xref:build-systems/cmake.adoc[].
-. xref:testing/continuous-integration.adoc[] tests.
+. xref:testing/continuous-integration.adoc[]
 
 The new library developer needs to consider all three, however the initial focus on a new library will be the first in this list. Or, to put it another way, "Quality assurance based on a wide range of targeted tests" as one of the key answers to Professor Hoare's question:
 
-"_How did software get so reliable without proof._"
+"_How did software get so reliable without proof?_"
 
-== Compiler Considerations
+== Define a Test Matrix
 
-Boost libraries rely on modern pass:[C++] features such as templates and the pass:[C++] Standard Library. Most modern compilers support those major features fairly well. But even today, years after the adoption of the pass:[C++] Standard, some compilers still don't support important minor features like partial template specialization.
+When you're looking to submit a library to the Boost collection, it's essential to ensure broad compatibility with various compilers, platforms, and configurations. Create a test matrix of what you intend to support, and document what you do not intend to support, and consider:
 
-Boost library authors often expend a great deal of effort trying to work around compiler deficiencies. Nevertheless, some libraries will not compile at all with certain compilers, or may have reduced or compromised functionality. Even if the current release of a compiler supports a Boost library, older versions of the compiler may produce code that does not work properly.
+. Testing with multiple versions of each compiler. Popular compilers to consider include:
+** GCC (GNU Compiler Collection)
+** Clang
+** MSVC (Microsoft Visual pass:[C++])
+** ICC (Intel pass:[C++] Compiler)
```

Comment:
I don't think most libraries are testing ICC for various reasons, although it's a popular compiler.

Other common compilers we test are MinGW, clang-cl, and apple-clang, but we don't really need to list them here because they are just variants of GCC and clang.